src/spatial/triangle.rs

```diff
@@ -0,0 +1,421 @@
+    /// pruning.
+    ///
+    /// Takes three vertices as `[y0, x0, y1, x1, y2, x2]` and edge boundary flags `[ab, bc, ca]`
+    /// indicating which edges are polygon boundaries. Returns a triangle struct with the bounding
+    /// box in the first four words as `[min_y, min_x, max_y, max_x]`. When decoded, the vertex
+    /// order may differ from the original input to `new()` due to normalized rotation.
+    pub fn new(doc_id: u32, triangle: [i32; 6], boundaries: [bool; 3]) -> Self {
+        let mut ay = triangle[0];
+        let mut ax = triangle[1];
+        let mut by = triangle[2];
+        let mut bx = triangle[3];
+        let mut cy = triangle[4];
+        let mut cx = triangle[5];
+        let mut ab = boundaries[0];
+        let mut bc = boundaries[1];
+        let mut ca = boundaries[2];
+        // rotate edges and place minX at the beginning
+        if bx < ax || cx < ax {
+            let temp_x = ax;
+            let temp_y = ay;
+            let temp_boundary = ab;
+            if bx < cx {
+                ax = bx;
+                ay = by;
+                ab = bc;
+                bx = cx;
+                by = cy;
+                bc = ca;
+                cx = temp_x;
+                cy = temp_y;
+                ca = temp_boundary;
+            } else {
+                ax = cx;
+                ay = cy;
+                ab = ca;
+                cx = bx;
+                cy = by;
+                ca = bc;
+                bx = temp_x;
+                by = temp_y;
+                bc = temp_boundary;
+            }
+        } else if ax == bx && ax == cx {
+            // degenerated case, all points with same longitude
+            // we need to prevent that ax is in the middle (not part of the MBS)
+            if by < ay || cy < ay {
+                let temp_x = ax;
+                let temp_y = ay;
+                let temp_boundary = ab;
+                if by < cy {
+                    ax = bx;
+                    ay = by;
+                    ab = bc;
+                    bx = cx;
+                    by = cy;
+                    bc = ca;
+                    cx = temp_x;
+                    cy = temp_y;
+                    ca = temp_boundary;
+                } else {
+                    ax = cx;
+                    ay = cy;
+                    ab = ca;
+                    cx = bx;
+                    cy = by;
+                    ca = bc;
+                    bx = temp_x;
+                    by = temp_y;
+                    bc = temp_boundary;
+                }
+            }
+        }
+        // change orientation if CW
+        if orient2d(
+            Coord { y: ay, x: ax },
+            Coord { y: by, x: bx },
+            Coord { y: cy, x: cx },
+        ) < 0.0
+        {
+            let temp_x = bx;
+            let temp_y = by;
+            let temp_boundary = ab;
+            // ax and ay do not change, ab becomes bc
+            ab = bc;
```

Comment:
I believe this is a bug:
```suggestion
            ab = ac;
```